### PR TITLE
testcafe configuration supports the `screenshots` option (closes #4322)

### LINKED ISSUE #4322: Support `screenshots` options as an object in configuration file

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/configuration/testcafe-configuration.ts [hunk 1]
```diff
@@ -136,6 +139,14 @@ export default class TestCafeConfiguration extends Configuration {
         filterOption.value = getFilterFn(filterOption.value) as Function;
     }
 
+    private _ensureScreenshotPath (): void {
+        const path = resolve(process.cwd(), 'screenshots');
```
Comment (miherlosev) [inline src/configuration/testcafe-configuration.ts]:
```js
import resolvePathRelativelyCwd from 'src/utils/resolve-path-relatively-cwd.js';

const DEFAULT_SCREENSHOTS_FOLDER = 'screenshots';
...
const path = resolvePathRelativelyCwd(DEFAULT_SCREENSHOTS_FOLDER);
```

File: src/configuration/configuration-base.ts [hunk 2]
```diff
@@ -190,8 +196,22 @@ export default class Configuration {
         option.source = source;
     }
 
+    protected _addOverriddenOptionIfRequired (value1: OptionValue, value2: OptionValue, source: OptionSource, optionName: string): void {
```
Comment (miherlosev) [inline src/configuration/configuration-base.ts]: `_addOverridenOptionIfNecessary` is more suitable function name.

File: src/runner/task.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ export default class Task extends AsyncEventEmitter {
         this.tests                   = tests;
         this.opts                    = opts;
         this.proxy                   = proxy;
-        this.screenshots             = new Screenshots(this.opts.screenshotPath, this.opts.screenshotPathPattern, this.opts.screenshotsFullPage);
+        this.screenshots             = new Screenshots(this.opts.screenshots, this.opts);
```
Comment (miherlosev) [inline src/runner/task.js]: It's a strange construction - `new Screenshots(this.opts.screenshots, this.opts);`
First, you pass to the constructor a part of options, then - the whole amount of options.
Simplify to `new Screenshots(this.opts);` or something else.

Comment (AlexKamaev) [reply src/runner/task.js]: the `Screenshots` constructor will not require `this.opts` parameter in future. This strange construction is only for backward compatibility, so we will definitely remove this in the future

File: src/configuration/testcafe-configuration.ts (hunk 1 above)
Comment (AlexKamaev) [reply src/configuration/testcafe-configuration.ts]: ok

File: src/configuration/configuration-base.ts (hunk 2 above)
Comment (AlexKamaev) [reply src/configuration/configuration-base.ts]: ok

File: src/runner/task.js (hunk 3 above)
Comment (miherlosev) [reply src/runner/task.js]: Ok

**Comment by AlexKamaev** [comment]
Well, I don't find it too complex. The `merging` is only one call to the `mergeWith` method and a predicate. And we do not need to collect any properties at all. Just one predicate.

**Pull request merged**